src/main/java/com/snowflake/kafka/connector/internal/BufferThreshold.java

```diff
@@ -159,27 +167,27 @@ private static boolean verifyBufferFlushTimeThreshold(
                   SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC,
                   providedFlushTimeSecondsInConfig,
                   thresholdTimeToCompare)));
-          return false;
+          return SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC;
         }
       } catch (NumberFormatException e) {
         LOGGER.error(
             Logging.logMessage(
                 "{} should be an integer. Invalid integer was provided:{}",
                 SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC,
                 providedFlushTimeSecondsInStr));
-        return false;
+        return SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC;
       }
     }
-    return true;
+    return "";
```

Comment:
not a big fan of returning an empty string here, can you return an ImmutablePair with empty things in it which can be a static final constant. (Denoting nothing is wrong with this config)

At the caller side, we can just filter that empty string to just have all invalid Pairs.